src/solc_0.8/defi/rules/RequirementsRules.sol

```diff
@@ -0,0 +1,204 @@
+        _listERC721Index[indexToDelete] = addrToMove;
+        _listERC721[addrToMove].index = indexToDelete;
+        _listERC721Index.pop();
+    }
+
+    function delRequireERC1155fromList(IERC1155 reqContract) external onlyOwner {
+        require(reqContract != IERC1155(address(0)), "RequirementsRules: invalid address");
+        require(_islistERC1155Member(reqContract), "RequirementsRules: contract is not in the list");
+        uint256 indexToDelete = _listERC1155[reqContract].index;
+        IERC1155 addrToMove = _listERC1155Index[_listERC1155Index.length - 1];
+        _listERC1155Index[indexToDelete] = addrToMove;
+        _listERC1155[addrToMove].index = indexToDelete;
+        _listERC1155Index.pop();
+    }
+
+    function _islistERC721Member(IERC721 reqContract) internal view returns (bool) {
+        if (_listERC721Index.length == 0) return false;
+        return (_listERC721Index[_listERC721[reqContract].index] == reqContract);
+    }
+
+    function _islistERC1155Member(IERC1155 reqContract) internal view returns (bool) {
+        if (_listERC721Index.length == 0) return false;
+        return (_listERC1155Index[_listERC1155[reqContract].index] == reqContract);
+    }
+
+    function _checkERC721List(address account) internal view returns (uint256) {
+        uint256 _totalBal = 0;
+        uint256 _maxStake = 0;
+
+        for (uint256 i = 0; i < _listERC721Index.length; i++) {
+            IERC721 reqContract = _listERC721Index[i];
+
+            if (_listERC721[reqContract].balanceOf == true) {
+                uint256 bal = reqContract.balanceOf(account);
+                _totalBal = _totalBal + bal;
+            } else {
+                for (uint256 j = 0; j < _listERC721[reqContract].ids.length; j++) {
+                    address owner = reqContract.ownerOf(_listERC721[reqContract].ids[j]);
+                    if (owner == account) {
+                        ++_totalBal;
+                    }
+                }
+            }
+
+            if (_totalBal < _listERC721[reqContract].reqAmount) {
+                _totalBal = 0;
+            }
+
+            _maxStake = _totalBal * _listERC721[reqContract].maxAmount;
+        }
+
+        return _maxStake;
+    }
+
+    function _checkERC1155List(address account) internal view returns (uint256) {
+        uint256 _totalBal = 0;
+        uint256 _maxStake = 0;
+
+        for (uint256 i = 0; i < _listERC1155Index.length; i++) {
+            IERC1155 reqContract = _listERC1155Index[i];
+
+            for (uint256 j = 0; j < _listERC1155[reqContract].ids.length; j++) {
+                uint256 bal = reqContract.balanceOf(account, _listERC1155[reqContract].ids[j]);
+
+                _totalBal = _totalBal + bal;
+            }
+
+            if (_totalBal < _listERC1155[reqContract].reqAmount) {
+                _totalBal = 0;
+            }
+
+            _maxStake = _totalBal * _listERC1155[reqContract].maxAmount;
+        }
+
+        return _maxStake;
+    }
+}
```

Comment:
requitmenrRules & ContributionRules share some methods, could we make a base class ?